azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java

```diff
@@ -139,14 +137,14 @@ private boolean allowTelemetry(final ConfigurableEnvironment env) {
         return env.getProperty(Constants.AZURE_KEYVAULT_ALLOW_TELEMETRY, Boolean.class, true);
     }
 
-    @PostConstruct
-    private void trackCustomEvent() {
+    private void sendTelemetry() {
```

Comment:
Look like it can be extracted as a util method?

Reply:
In fact it is trade-off here. The `properties` for each starter is quit different, like b2c has `tenant`, storage has `storageAccount`. If extracts util method, only `sender` and `send` can be put into that method, you still to create the `properties` outside and call that util method.
From my person option, that may not make code less or clean.

Reply:
I see.